### PR TITLE
Flux latents fix

### PR DESCRIPTION
**Pull request by DN6**
# What does this PR do?
This PR: #9711 changed the latent preparation step so that the `vae_scale_factor` and default sample size are more clear. 

However, we overlooked the fact that this operation (with scale_factor == 16)

```python
        height = 2 * (int(height) // self.vae_scale_factor)
        width = 2 * (int(width) // self.vae_scale_factor)
```

Is not equivalent to this operation with scale_factor == 8
```python
height = int(height) // self.vae_scale_factor
```
For resolutions that are divisible by 8 but not by 16 the current pipeline on main will error out, because the implementation does not account for the fact that the latent width and height must be divisible by 2 for the packing step to work.

The old implementation scaled the height and width by `//16` and then multiplied by 2. This results in slightly resized images if the height or width aren't multiples of 16.

This PR:
1. Adds steps to account for latent packing. Which will result in resized output images (like we had before)
2. Adds a warning that the image will be resized if an incompatible image is provided
3. Adds fast tests to check Flux with expected and unexpected image shapes to verify that the scaling is properly applied.      

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by bghira** [comment]
this seems really important but considering how slowly things move around here, not so sure it will be merged in a timely manner.

File: src/diffusers/pipelines/flux/pipeline_flux.py [hunk 1]
```diff
@@ -501,8 +505,10 @@ def prepare_latents(
         generator,
         latents=None,
     ):
-        height = int(height) // self.vae_scale_factor
-        width = int(width) // self.vae_scale_factor
+        # VAE applies 8x compression on images but we must also account for packing which requires
+        # latent height and width to be divisible by 2.
+        height = int(height) // self.vae_scale_factor - ((int(height) // self.vae_scale_factor) % 2)
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/flux/pipeline_flux.py:510]: is this same as this? (basically revert back to the original code but divisible by `self.vae_scale_factor*2`
a little bit easier to understand, no?
```python
height = 2 * (int(height) // (self.vae_scale_factor*2))
```

**Review by yiyixuxu — APPROVED** [review]
oh thanks for the fix!
I made a comment

**Comment by DN6** [comment]
Failing LoRA tests are unrelated. Merging.

**Pull request merged**